sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/implementation/models/BlobChangefeedEventData.java

```diff
@@ -0,0 +1,228 @@
+    private final String requestId;
+    private final String eTag;
+    private final String contentType;
+    private final Long contentLength;
+    private final BlobType blobType;
+    private final Long contentOffset;
+    private final String destinationUrl;
+    private final String sourceUrl;
+    private final String blobUrl;
+    private final boolean recursive;
+    private final String sequencer;
+
+    /**
+     * Constructs a {@link BlobChangefeedEventData}.
+     *
+     * @param api The api.
+     * @param clientRequestId The client request id.
+     * @param requestId The request id.
+     * @param eTag The eTag.
+     * @param contentType The content type.
+     * @param contentLength Th4e content length.
+     * @param blobType {@link BlobType}
+     * @param contentOffset The content offset.
+     * @param destinationUrl The destination url.
+     * @param sourceUrl The source url.
+     * @param blobUrl The blob url.
+     * @param recursive Whether or not this operation was recursive.
+     * @param sequencer The sequencer.
+     */
+    public BlobChangefeedEventData(String api, String clientRequestId, String requestId, String eTag,
+        String contentType, Long contentLength, BlobType blobType, Long contentOffset, String destinationUrl,
+        String sourceUrl, String blobUrl, boolean recursive, String sequencer) {
+        this.api = api;
+        this.clientRequestId = clientRequestId;
+        this.requestId = requestId;
+        this.eTag = eTag;
+        this.contentType = contentType;
+        this.contentLength = contentLength;
+        this.blobType = blobType;
+        this.contentOffset = contentOffset;
+        this.destinationUrl = destinationUrl;
+        this.sourceUrl = sourceUrl;
+        this.blobUrl = blobUrl;
+        this.recursive = recursive;
+        this.sequencer = sequencer;
+    }
+
+    static BlobChangefeedEventData fromRecord(Object d) {
+        AvroSchema.checkType("data", d, Map.class);
+        Map<?, ?> data = (Map<?, ?>) d;
+
+        if (!data.get(AvroConstants.RECORD).equals("BlobChangeEventData")) {
+            throw new IllegalArgumentException("Not a valid BlobChangefeedEventData.");
+        }
+
+        Object api = data.get("api");
+        Object clientRequestId = data.get("clientRequestId");
+        Object requestId = data.get("requestId");
+        Object eTag = data.get("etag");
+        Object contentType = data.get("contentType");
+        Object contentLength = data.get("contentLength");
+        Object blobType = data.get("blobType");
+        Object contentOffset = data.get("contentOffset");
+        Object destinationUrl = data.get("destinationUrl");
+        Object sourceUrl = data.get("sourceUrl");
+        Object blobUrl = data.get("url");
+        Object recursive = data.get("recursive");
+        Object sequencer = data.get("sequencer");
+
+        return new BlobChangefeedEventData(
+            BlobChangefeedEvent.nullOrString("api", api),
```

Comment:
If these are used across various models then they should exist in separate class. For example ChangeFeedModelValidator or something like that.

Reply:
@kasobol-msft Can you take another look?